### PR TITLE
conan: Migrate to conan v2 tools

### PR DESCRIPTION
**Pull request by hellozee**
Tried to use the new Autotools build tools provided by conan v2 but somehow it never set the proper include directories or the lib directories. Beside that, didn't find any example on conan center index to understand how to use this properly (though it is highly likely that I may have missed some).

### CODE REVIEW

**Comment by sorny92** [comment]
Can you put it to draft until it's green or you need help?

File: conanfile.py [hunk 1]
```diff
@@ -172,15 +192,17 @@ def __init__(self, conanfile, prefix_dir):
     def _get_binaries(self, dest_dir):
         """Get the binaries from the special embeddable Python package"""
         url = "https://www.python.org/ftp/python/{0}/python-{0}-embed-amd64.zip"
-        tools.get(url.format(self.conanfile.pyversion), destination=dest_dir)
+        get(self.conanfile, url.format(self.conanfile.pyversion), destination=dest_dir)
```
Comment (sorny92) [inline conanfile.py:195]:
```suggestion
        get(self, url.format(self.conanfile.pyversion), destination=dest_dir)
```
Can be like that?

File: conanfile.py [hunk 2]
```diff
@@ -172,15 +192,17 @@ def __init__(self, conanfile, prefix_dir):
...
 
     def _get_headers_and_lib(self, dest_dir):
         """We also need headers and the `python3.lib` file to link against"""
         url = f"https://www.python.org/ftp/python/{self.conanfile.pyversion}/amd64/dev.msi"
-        tools.download(url, filename="tmp\\dev.msi")
+        download(self.conanfile, url, filename="tmp\\dev.msi")
```
Comment (sorny92) [inline conanfile.py:200]:
```suggestion
        download(self, url, filename="tmp\\dev.msi")
```
same as before

File: conanfile.py [hunk 3]
```diff
@@ -172,15 +192,17 @@ def __init__(self, conanfile, prefix_dir):
...
         build_folder = self.conanfile.build_folder
-        self.conanfile.run(f"msiexec.exe /a {build_folder}\\tmp\\dev.msi targetdir={dest_dir}")
-        tools.rmdir("tmp")
+        self.conanfile.run(
+            f"msiexec.exe /a {build_folder}\\tmp\\dev.msi targetdir={dest_dir}"
+        )
+        rmdir(self.conanfile, "tmp")
```
Comment (sorny92) [inline conanfile.py:205]:
```suggestion
        rmdir(self, "tmp")
```
eghm

File: test_package/conanfile.py [hunk 4]
```diff
@@ -27,8 +27,11 @@ def configure(self):
 
     def imports(self):
         import embedded_python_tools
+
         embedded_python_tools.symlink_import(self, dst="bin/python")
-        self.copy("licenses/*", dst="licenses", folder=True, ignore_case=True, keep_path=False)
+        self.copy(
```
Comment (sorny92) [inline test_package/conanfile.py:32]:
```suggestion
        copy(self,...
```

**Review by sorny92 — COMMENTED** [review]
Haven't test so I don't have more comment for now.

File: conanfile.py (hunk 1 above)
Comment (hellozee) [reply conanfile.py:196]: No it needs to be a conanfile, 😓

**Comment by hellozee** [comment]
I tried my best, couldn't figure out how to change the `self.copy` to `copy`

File: .github/workflows/test_package.yml [hunk 5]
```diff
@@ -24,7 +24,7 @@ jobs:
         echo CXX=/usr/bin/clang++ >> $GITHUB_ENV
     - name: Install Conan
       run: | 
-        python -m pip install conan==1.53
+        python -m pip install conan==1.59
```
Comment (sorny92) [inline .github/workflows/test_package.yml:27]: Oh nice

**Review by sorny92 — COMMENTED** [review]
Same as aqt. Have you tested with lumicks/stable? 

If that's the case I would say to create a PR in the consumers to modify it and we can migrate if green? Once is green we can merge this.

File: changelog.md [hunk 6]
```diff
@@ -1,8 +1,9 @@
 # Changelog
 
-## v1.4.6 | In progress
+## v1.4.6 | 2023-02-28
```
Comment (hellozee) [inline changelog.md:3]: goddammit forgot to change the version

**Review by sorny92 — APPROVED** [review]
I have no more code comment. Looks good to me. If boilerplate is your friend and nothing breaks 🚢

**Pull request merged**